src/substrait-to-d3.js

```diff
@@ -0,0 +1,296 @@
+        processedLinks.push({ source, target });
+    }
+    return processedLinks;
+}
+
+// Specifying color of graph nodes
+function nodeColor(nodeType) {
+    switch (nodeType) {
+        case 'root':
+            return "#4D2205";
+        case 'project':
+            return "#0E4D05";
+        case 'read':
+            return "#052748";
+        case 'join':
+            return '#46054D'
+    }
+}
+
+// Specifying Font-Awesome icons for graph nodes
+function nodeIcon(nodeType) {
+    switch (nodeType) {
+        case 'root':
+            return '\uf0ce';
+        case 'project':
+            return '\uf12b';
+        case 'read':
+            return '\uf1c0';
+        case 'join':
+            return '\uf126';
+    }
+}
+
+function typeToLabel(nodeType) {
+    if (nodeType == 'root') {
+        return 'Sink';
+    } else {
+        return nodeType[0].toUpperCase() + nodeType.substring(1);
+    }
+}
+
+// Drawing Graph using d3JS
+function drawGraph(pre_nodes, pre_links) {
+    let width = 960,
+        height = 375;
+
+    let nodes = processNodes(pre_nodes);
+    let links = processEdges(nodes, pre_links);
+
+    // Instantiating a Force Simulation
+    var force = d3.forceSimulation(nodes)
+        .force('charge', d3.forceManyBody().strength(-100))
+        .force('center', d3.forceCenter(width / 2, height / 2))
+        .force('link', d3.forceLink().links(links))
+        .force('collide', d3.forceCollide(d => 55))
+
+    var svg = d3.select("svg")
+        .attr("width", width)
+        .attr("height", height);
+
+    // For arrowheads in edges
+    svg.append('defs').append('marker')
+        .attr("id", 'arrowhead')
+        .attr('viewBox', '-0 -5 10 10')
+        .attr('refX', 27)
+        .attr('refY', 0)
+        .attr('orient', 'auto')
+        .attr('markerWidth', 13)
+        .attr('markerHeight', 13)
+        .attr('xoverflow', 'visible')
+        .append('svg:path')
+        .attr('d', 'M 0,-5 L 10 ,0 L 0,5')
+        .attr('fill', 'black')
+        .style('stroke', 'none');
+
+    // specifying links in the graph
+    const link = svg.append("g")
+        .selectAll("path")
+        .data(links)
+        .join("path")
+        .attr('marker-end', 'url(#arrowhead)')
+        .style("stroke", "black");
+
+    // specifying nodes in graph
+    const node = svg.append("g")
+        .selectAll("g")
+        .data(nodes)
+        .join("g")
+        .call(drag(force));
+
+    // specifying node shape and color
+    node.append("circle")
+        .attr("r", 25)
+        .style("fill", (d) => {
+            return nodeColor(d.type);
+        }).style("stroke-opacity", 0.3);
+
+    // adding icon in node
+    node.append("text")
+        .attr('class', 'fa')
+        .attr("x", -3.5)
+        .attr("y", 3)
+        .attr("dx", "-0.15em")
+        .attr("dy", "0.2em")
+        .text(function (d) {
+            return nodeIcon(d.type);
+        });
+
+
+    // displaying node data on click
+    node.on("click", function (d) {
```

Comment:
This works for now.  Eventually we will want to support something like...

```
drawGraph(graph['nodes'], graph['edges'], (nodeSelected) => {
  onNodeSelected(nodeSelected);
);
```

Where `onNodeSelected` is some callback provided by the caller of `drawGraph`.  This separates the concerns of table rendering (which d3 is not great at) from graph drawing (which d3 excels at).